src/Razor/src/Microsoft.VisualStudio.Editor.Razor/RazorTextViewConnectionListener.cs

```diff
@@ -85,10 +90,9 @@ public void SubjectBuffersDisconnected(ITextView textView, ConnectionReason reas
                     throw new ArgumentNullException(nameof(subjectBuffers));
                 }
 
-                _joinableTaskContext.Factory.Run(async () => {
-                    _joinableTaskContext.AssertUIThread();
-                    await _documentManager.OnTextViewClosedAsync(textView, subjectBuffers);
-                });
+                _joinableTaskContext.AssertUIThread();
```

Comment:
💡 Asserting the current thread in an asynchronous method is an anti-pattern. This should either be a synchronous method that asserts its affinity, or an asynchronous method that switches on demand.